message/message.go

```diff
@@ -0,0 +1,105 @@
+package message
+
+import "strings"
+
+// Message represents DNS messages
+type Message struct {
+	header  *header
+	queries []*query
+	Answers []*Record
+}
+
+type header struct {
+	ID      uint16
+	QR      byte
```

Comment:
Who would need to know details outside the Message object about the header? 

Also, bytes are unsigned integers.